adapta/storage/blob/s3_storage_client.py

```diff
@@ -0,0 +1,145 @@
+        :param blob_path: Blob path in DataPath notation.
+        :param metadata: Optional blob tags or metadata to attach.
+        :param overwrite: whether a blob should be overwritten or an exception thrown if it already exists.
+        :param serialization_format: The serialization format.
+            The type (T) of the serialization format must be compatible with the provided data.
+        :return:
+        """
+        bytes_ = serialization_format().serialize(data)
+        self.bucket.put_object(cast_path(blob_path).path, bytes_)
+
+    def delete_blob(self, blob_path: DataPath) -> None:
+        """
+        Deletes blob at blob_path
+
+        :param blob_path: Blob path as DataPath object
+        """
+        self.bucket.Object(blob_path.path).delete()
+
+    def list_blobs(
+        self, blob_path: DataPath, filter_predicate: Optional[Callable[[...], bool]] = None
+    ) -> Iterator[DataPath]:
+        """
+        Lists blobs in blob_path
+
+        :param blob_path: Blob path as DataPath object
+        :param filter_predicate: Take only blobs that match a supplied predicate.
+        :return: An iterator of DataPaths to blobs
+        """
+        return self.bucket.objects.all()
+
+    def read_blobs(
+        self,
+        blob_path: DataPath,
+        serialization_format: Type[SerializationFormat[T]],
+        filter_predicate: Optional[Callable[[...], bool]] = None,
+    ) -> Iterator[T]:
+        """
+         Reads data under provided path into the given format.
+
+        :param blob_path: Path to blob(s).
+        :param serialization_format: Format to deserialize blobs into.
+        :param filter_predicate: Take only blobs that match a supplied predicate.
+        :return: An iterator over deserialized blobs
+        """
+        for blob in self.bucket.objects.filter(Prefix=cast_path(blob_path).path):
+            if filter_predicate is not None and not filter_predicate(blob):
+                continue
+            yield serialization_format().deserialize(blob.get()["Body"].read())
+
+    def download_blobs(
+        self,
+        blob_path: DataPath,
+        local_path: str,
+        threads: Optional[int] = None,
+        filter_predicate: Optional[Callable[[...], bool]] = None,
+    ) -> None:
+        """
+        Not implemented in S3 Client
+        """
+        raise NotImplementedError("Not implemented in S3StorageClient")
+
+    def copy_blob(self, blob_path: DataPath, target_blob_path: DataPath, doze_period_ms: int) -> None:
+        """
+        Copy blob at `blob_path` to `target_blob_path`
+
+        :param blob_path: Path to source blob.
+        :param target_blob_path: Path to target blob.
+        :param doze_period_ms: number of ms to doze between polling the status of the copy.
+        """
+        copy_source = {"Bucket": self.bucket_name, "Key": blob_path.path}
+        self.bucket.Object(cast_path(blob_path).path).copy(copy_source, cast_path(target_blob_path).path)
```

Comment:
this is incorrect as this method copies using remote copy operation, your copy here is copy-through-local operation